src/parse.rs

```diff
@@ -160,11 +170,78 @@ pub fn get_mount_from<T, U>(target: T, iter: MountIter<U>)
     Ok(ret)
 }
 
+pub fn get_mount_search_from<U>(search: &Search, iter: MountIter<U>)
+        -> Result<Option<Vec<MountEntry>>, ParseError> where U: BufRead {
+    let mut ret = Vec::new();
+    for mount in iter {
+        match mount {
+            Ok(m) => {
+                match search {
+                    &Search::Spec(ref spec) => {
+                        if *spec == m.spec {
+                            ret.push(m);
+                        }
+                    },
+                    &Search::File(ref file) => {
+                        if *file == m.file {
+                            ret.push(m);
+                        }
+                    },
+                    &Search::Vfstype(ref vfstype) => {
+                        if *vfstype == m.vfstype {
+                            ret.push(m);
+                        }
+                    },
+                    &Search::Mntopts(ref mntops) => {
+                        // All the opts must be present for a match
+                        let count = mntops.len();
+                        let mut match_count = 0;
+
+                        // I'm pretty sure there is a more elegant way to do this
```

Comment:
Unless there is a compelling reason to avoid using HashSet I'ld do subset inclusion...which as far as I can tell is all this is.